packages/wallets/src/components/Base/WalletAlertMessage/WalletAlertMessage.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { ReactNode } from 'react';
+import useDevice from '../../../hooks/useDevice';
+import ErrorCircleCrossmark from '../../../public/images/error-circle-crossmark.svg';
+import InfoCircleDots from '../../../public/images/info-circle-dots.svg';
+import SuccessCircleCheckmark from '../../../public/images/success-circle-checkmark.svg';
+import { WalletButton, WalletText } from '../index';
+import styles from './WalletAlertMessage.module.css';
+
+const typeMapper = {
+    error: {
+        color: 'error',
+        icon: ErrorCircleCrossmark,
+    },
+    info: {
+        color: 'blue',
+        icon: InfoCircleDots,
+    },
+    success: {
+        color: 'success',
+        icon: SuccessCircleCheckmark,
+    },
+};
+
+type TAlertMessageProps =
+    | {
+          buttonLabel: ReactNode;
+          message: ReactNode;
+          onClickHandler: VoidFunction;
+          type: 'error' | 'info' | 'success';
+          variant: 'with-action-button';
+      }
+    | {
+          message: ReactNode;
+          type: 'error' | 'info' | 'success';
+          variant: 'base';
+      };
```

Comment:
```suggestion
type TAlertMessageProps = {
      message: ReactNode;
      type: 'error' | 'info' | 'success';
      variant: 'base';
      children: ReactNode;
};
```
A more better component design would be allowing the user to pass an action button as a `children` in my opinion. The `children` already have conditional check (if passed only show) and it allows flexibility for the consumer to pass their own `onClickHandler` without explicitly restricting to `VoidFunction` type.

With my proposed code review, consuming this `WalletAlertMessage` would look like this:

```
<WalletAlertMessage><WalletButton onClick={() => setSomething(true)} size='sm' text="Set Something" /></WalletAlertMessage>
```

Which looks way usable if you ask me